Geometry_Engine/Create/PlanarSurface.cs

```diff
@@ -79,7 +79,96 @@ public static PlanarSurface PlanarSurface(ICurve externalBoundary, List<ICurve>
                 }
             }
 
-            return new PlanarSurface { ExternalBoundary = externalBoundary, InternalBoundaries = internalBoundaries };
+            bool nonExistantCheckWarning = false;
+
+            try
+            {
+                if (externalBoundary.IIsSelfIntersecting())
+                {
+                    Reflection.Compute.RecordError("The provided externalBoundary is selfintersecting.");
+                    return null;
+                }
+            } catch
+            {
+                nonExistantCheckWarning = true;
+            }
+
+            try
+            {
+                if (internalBoundaries.Any(x => x.IIsSelfIntersecting()))
+                {
+                    Reflection.Compute.RecordError("At least one of the internalBoundaries is selfintersecting.");
```

Comment:
Can we remove the self-intersecting inner boundaries and build the planar surface only from the valid ones?

Reply:
We could, but it feels like a less clear-cut solution than say the boolean union as it would modify what the user might have meant, but I'm not opposed per say

Reply:
Same as above, leaning towards @epignatelli